gozstd.go

```diff
@@ -185,10 +185,30 @@ func Decompress(dst, src []byte) ([]byte, error) {
 	return DecompressDict(dst, src, nil)
 }
 
+// DecompressLimited appends decompressed src to dst and returns the result.
+//
+// If dst has insufficient capacity for decompressed src, allocated memory is limited by maxAlloc.
```

Comment:
The semantics of the `maxAlloc` arg is too fragile: it doesn't limit the size of the decompressed data if it fits the `dst` capacity. This may lead to subtle bugs at the caller side which expects that the `maxAlloc` unconditionally limits the size of the decompressed block. So it is better to rename the `maxAlloc` argument to `limit` and change its' semantics, so it always limits the size of the decompressed data.